test/test_trixi.jl

```diff
@@ -98,3 +97,41 @@ macro test_nowarn_debug(expr)
     end
   end
 end
+
+# Modified version of `@test_nowarn` that prints the content of `stderr` when
+# it is not empty and ignnores module replacements.
+macro test_nowarn_mod(expr)
+  quote
+    let fname = tempname()
+      try
+        ret = open(fname, "w") do f
+          redirect_stderr(f) do
+            $(esc(expr))
+          end
+        end
+        stderr_content = read(fname, String)
+        if !isempty(stderr_content)
+          println("Content of `stderr`:\n", stderr_content)
```

Comment:
Why print ``content of `stderr:` `` and not just the same stuff that would've been printed with `@test_nowarn`?

Reply:
Well, I find the error message
```
Test Failed at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:671
  Expression: contains_warn(read(fname, String), $(Expr(:escape, :(r"^(?!.)"s))))
Stacktrace:
 [1] macro expansion
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:671 [inlined]
 [2] macro expansion
   @ ~/.julia/dev/Trixi/test/test_trixi.jl:35 [inlined]
 [3] macro expansion
   @ REPL[3]:10 [inlined]
 [4] macro expansion
   @ /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.6/Test/src/Test.jl:1151 [inlined]
 [5] top-level scope
   @ REPL[3]:10
```
highly non-helpful since I don't get any hint what kind of warning appeared :shrug: Thus, I wanted to print possible warnings to help us debug possible problems

Reply:
Yeah, I like the verbosity and the more helpful error message here. Though you could argue that you should've already seen the actual warning message in the test output before, this makes it just a little easier to debug.

Reply:
> Though you could argue that you should've already seen the actual warning message in the test output before

No, you can't see the error message or warning since `@test_nowarn` (and this variant`) redirect `stderr` output to a temporary file invisible to the user.

Reply:
Ah right. OK, in this case, I strongly argue that we need this new functionality, so feel free to resolve this conversation and merge once the tests pass!